eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -809,20 +816,39 @@ object Router {
     * @param ignoredEdges a set of extra edges we want to IGNORE during the search
     * @return the computed route to the destination @targetNodeId
     */
-  def findRoute(g: DirectedGraph, localNodeId: PublicKey, targetNodeId: PublicKey, amountMsat: Long, numRoutes: Int, extraEdges: Set[GraphEdge] = Set.empty, ignoredEdges: Set[ChannelDesc] = Set.empty): Try[Seq[Hop]] = Try {
+  def findRoute(g: DirectedGraph,
+                localNodeId: PublicKey,
+                targetNodeId: PublicKey,
+                amountMsat: Long,
+                numRoutes: Int,
+                extraEdges: Set[GraphEdge] = Set.empty,
+                ignoredEdges: Set[ChannelDesc] = Set.empty,
+                routeParams: RouteParams = DEFAULT_ROUTE_PARAMS): Try[Seq[Hop]] = Try {
+
     if (localNodeId == targetNodeId) throw CannotRouteToSelf
 
-    val foundRoutes = Graph.yenKshortestPaths(g, localNodeId, targetNodeId, amountMsat, ignoredEdges, extraEdges, numRoutes).toList match {
-      case Nil => throw RouteNotFound
-      case route :: Nil if route.path.isEmpty => throw RouteNotFound
-      case foundRoutes => foundRoutes
+    val boundaries: RichWeight => Boolean = { weight =>
+      ((weight.cost - amountMsat) < routeParams.maxFeeBaseMsat || (weight.cost - amountMsat) < (routeParams.maxFeePct * amountMsat)) &&
+      weight.length <= routeParams.routeMaxLength && weight.length <= DEFAULT_ROUTE_MAX_LENGTH &&
+      weight.cltv <= routeParams.routeMaxCltv
     }
 
-    // minimum cost
-    val minimumCost = foundRoutes.head.weight
+    val foundRoutes = Graph.yenKshortestPaths(g, localNodeId, targetNodeId, amountMsat, ignoredEdges, extraEdges, numRoutes, boundaries).toList match {
+      case Nil =>
+        if(routeParams.routeMaxLength < DEFAULT_ROUTE_MAX_LENGTH){
+          return findRoute(g, localNodeId, targetNodeId, amountMsat, numRoutes, extraEdges, ignoredEdges, routeParams.copy(routeMaxLength = DEFAULT_ROUTE_MAX_LENGTH))
+        } else {
+          throw RouteNotFound
+        }
+      case route :: Nil if route.path.isEmpty =>
```

Comment:
It would be cleaner to not return an empty route here and have a single "cannot find a route" check

Reply:
done 05b5015